inwhale/observers/mse.py

```
@@ -0,0 +1,53 @@
+
+class MSEObserver(Observer):
+    def __init__(self, bits, num_candidates=100):
+        super().__init__()
+        self.bits = bits
+        self.num_candidates = num_candidates
+        self.min_val = None
+        self.max_val = None
+
+    def observe(self, x):
+
+        if abs_max == 0:
```

Ruff (0.14.8)

Undefined name `abs_max`

(F821)